test/serializer/abstract/ModelFunction.js

```diff
@@ -0,0 +1,15 @@
+// add at runtime: global.fun = console.log;
+var result = [];
+global.log = function (arg) { result.push(arg); };
+if (global.__assumeDataProperty) {
+  __assumeDataProperty(global, "fun", 
```

Comment:
I can change the names to SKIP_INVARIANT, FULL_INVARIANT, VALUE_DEFINED_INVARIANT to make it more clear from the argument passed.